keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigYamlParsingFilterTests.kt

```diff
@@ -0,0 +1,126 @@
+package com.netflix.spinnaker.keel.rest
+
+import com.fasterxml.jackson.databind.jsontype.NamedType
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.netflix.spinnaker.keel.api.ec2.EC2_SECURITY_GROUP_V1
+import com.netflix.spinnaker.keel.api.ec2.SecurityGroupSpec
+import com.netflix.spinnaker.keel.api.plugins.SupportedKind
+import com.netflix.spinnaker.keel.core.api.SubmittedDeliveryConfig
+import com.netflix.spinnaker.keel.core.api.SubmittedResource
+import com.netflix.spinnaker.keel.ec2.jackson.registerKeelEc2ApiModule
+import com.netflix.spinnaker.keel.test.configuredTestObjectMapper
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import javax.servlet.FilterChain
+import javax.servlet.ServletRequest
+import javax.servlet.http.HttpServletRequestWrapper
+import org.springframework.mock.web.MockHttpServletRequest
+import org.springframework.mock.web.MockHttpServletResponse
+import strikt.api.expectThat
+import strikt.assertions.hasSize
+import strikt.assertions.isA
+import strikt.assertions.isEqualTo
+
+class DeliveryConfigYamlParsingFilterTests : JUnit5Minutests {
+  object Fixture {
+    val chain: FilterChain = mockk()
+    val filter = DeliveryConfigYamlParsingFilter()
+    val request = MockHttpServletRequest("POST", "/delivery-configs/")
+    val response = MockHttpServletResponse()
+    val objectMapper = configuredTestObjectMapper().registerKeelEc2ApiModule()
+
+    init {
+      objectMapper.registerSubtypes(EC2_SECURITY_GROUP_V1.toNamedType())
+      request.contentType = "application/x-yaml"
+      request.setContent("""
+        application: nimakspin
+        serviceAccount: myservice@keel.io
+        artifacts: []
+        hasManagedResources: true
+        environments:
+          - name: testing
+            hasManagedResources: true
+            resources:
+            -
+              kind: ec2/security-group@v1
+              spec: &secgrp
+                moniker:
+                  app: nimakspin
+                locations:
+                  account: my-aws-account
+                  vpc: my-vpc
+                  regions:
+                  - name: us-west-2
+                description: "test LoadBalancer securityGroup by ALB Ingress Controller"
+                inboundRules:
+                - protocol: TCP
+                  portRange:
+                    startPort: 80
+                    endPort: 80
+                  blockRange: "0.0.0.0/0"
+                - protocol: TCP
+                  portRange:
+                    startPort: 22
+                    endPort: 22
+                  blockRange: "0.0.0.0/0"
+          - name: prod
+            hasManagedResources: true
+            resources:
+            - kind: ec2/security-group@v1
+              spec:
+                << : *secgrp
```

Comment:
@nimakaviani Note that this is slightly different than your example because we do additional validation of the spec and verify that resource IDs (which derive from the `moniker` field) are unique. IMO this is a good-enough compromise, but wanted to give you a heads-up.

Reply:
looks great thanks! 

agreed. parsing is independent of the validation. As long as parsing works as expected, plugins can to decide at what level to use anchors and aliases and how to do the validations. so good enough IMO too. thanks again.